packages/x-data-grid/src/models/props/DataGridProps.ts

```diff
@@ -251,14 +251,6 @@ export interface DataGridPropsWithDefaultValues<R extends GridValidRowModel = an
    * @default false
    */
   ignoreDiacritics: boolean;
-  /**
-   * If `select`, a group header checkbox in indeterminate state (like "Select All" checkbox)
-   * will select all the rows under it.
-   * If `deselect`, it will deselect all the rows under it.
-   * Works only if `checkboxSelection` is enabled.
-   * @default "deselect"
```

Comment:
Do we plan to deprecate on v7 next patch?

Reply:
Seems a good idea to me. I'd be in favor of doing so, to let the v7 users mentally prepare for the breaking change beforehand.

Reply:
Created https://github.com/mui/mui-x/issues/15614 to track it